[CS2113-F14-2] DinEZ

Zhengwinter opened the pull request:
The application is an integrated system solution which aims to help operations manager address all aspects of restaurant management that could be encountered during the daily running of a restaurant in a digitalized way, including order taking, delivery tracking, inventory monitoring and customer feedback etc.

adamzzq committed:
Merge pull request #82 from adamzzq/branch-print-receipt

Implement getReceipt method in Order.java

adamzzq committed:
Implement Order.getReceipt method with discount

adamzzq committed:
Merge pull request #84 from adamzzq/branch-print-receipt

Implement Order.getReceipt method with discount

Zhengwinter committed:
Implement Menu commands

Zhengwinter committed:
Merge branch 'master' into master

Zhengwinter committed:
Restructure the complete command to be used by both order and menu logic

Zhengwinter committed:
Resolve merge conflicts

Zhengwinter committed:
Reformat code to comply with char limit

Zhengwinter committed:
Merge pull request #85 from Zhengwinter/master

Implement Menu commands (MenuAdd, MenuDelete, MenuComplete)

Zhengwinter committed:
Refactor code and abstract out classes, format in CamelCase

Zhengwinter committed:
Merge pull request #86 from Zhengwinter/master

Refactor code and abstract out classes, format in CamelCase

Xb990219 committed:
Update V2.0 working draft

Xb990219 committed:
Merge branch 'Xb990219-mainLogic'

Xb990219 committed:
resolve merge conflict

Xb990219 committed:
Fix minor bugs

Xb990219 committed:
Fix checkstyle

Xb990219 committed:
Merge pull request #88 from Xb990219/master

Update V2.0

ThawTunZan reviewed (commented):
Add a example output for each commands

ThawTunZan reviewed (commented):
There is "OrderLogic", but may want to include the Logic for other actions like remove or view

ThawTunZan reviewed (commented):
May want an explanation for some of the important methods in each class

ThawTunZan reviewed (commented):
May want to make it neater with the headers to make it easier to read

webtjs committed:
Update User Guide for v2.0

webtjs committed:
Merge pull request #91 from webtjs/master

Update User Guide for v2.0

adamzzq committed:
Update UG with input formats specification

adamzzq committed:
Merge pull request #92 from adamzzq/branch-UG

Update UG with input formats specification

Zhengwinter committed:
Add FAQ in the user guide

Zhengwinter committed:
Merge pull request #93 from Zhengwinter/master

Add FAQ in the user guide

adamzzq committed:
Include confirmation message to OrderCompleteCommand

adamzzq committed:
Merge pull request #94 from adamzzq/branch-complete-order-msg

Include confirmation message to OrderCompleteCommand

Xb990219 committed:
Add confirmation command for completing an order

Xb990219 committed:
Merge pull request #95 from Xb990219/master

Add confirmation command for completing an order

royaliceblue commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```
Good that the acknowledgment section is here. But will it be better to update the section?

royaliceblue commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![Architecture](images\Architecture.png)
```
An excellent high-level class diagram! Simple, clear and straight to the point. Good for understanding.

royaliceblue commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues 
+the command `Create order -menu 01`, `add -item 001` and `complete`.
+
+![Sequence Diagram](images\ArchitectureSequanceDiagram.png)
```
Clear sequence diagram.

<img width="865" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/48584322/a5e34340-0aea-4869-9bb1-a096152c8e83">

Will it be better is there is a better contrast of the text and the colour of the background for better readability?

royaliceblue commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues 
+the command `Create order -menu 01`, `add -item 001` and `complete`.
+
+![Sequence Diagram](images\ArchitectureSequanceDiagram.png)
```
<img width="359" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/48584322/da78e83b-95c7-45b7-ab7d-3b9ec3e20611">

Will it be better for the text to not be above the diagram lines?

NGXZS replied:
It may be clearer if we can highlight the commands `Create order -menu 01`, `add -item 001` and `complete` in the sequence diagram.
but overall, i like the colors for readability and it looks neat

royaliceblue commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
+The *Class Diagram* below shows how the model components interact with each other, including interactions such as 
+dependencies, associations and inheritance.
+
+![Class Diagram](images/modelcomponent.png)
```
<img width="480" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/48584322/ec66e95e-aeaa-44f4-abbb-dde3d94f34e0">

Good class diagram! But will it be better to leave the getters and setters out of the diagram since they are not necessary?

NGXZS replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110776396/ac153433-6bbb-4f9f-be45-db65b514c896)
we can also consider removing other unimportant functions here too

royaliceblue commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
+The *Class Diagram* below shows how the model components interact with each other, including interactions such as 
+dependencies, associations and inheritance.
+
+![Class Diagram](images/modelcomponent.png)
```
<img width="212" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/48584322/5f43432b-ae8b-4f50-8c8b-60728e18fd4a">

Will it be better to remove the icon as it is not taught in class? It can be disabled 😄

royaliceblue reviewed (commented):
Overall a great DG! Great simplified diagrams, straight to the point and clear. Just a few nits that can be worked on, especially removing the icon from the class diagram.

webtjs committed:
MainCommandTest.java: Refactor test code

webtjs committed:
Merge pull request #96 from webtjs/branch-v2.0

Add tag for v2.0, refactor code

NGXZS commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
+
+The app's work is done by the following components:
+
+* [**`MainLogic`**](#sublogic-component): The main logic command executor.
+* [**`SubLogic`**](#sublogic-component): The sub logics(i.e. **`MenuLogic`**, **`OrderLogic`**) command executor.
+* [**`Parser`**](#parser-component): The parser that parses user input to command.
+* [**`Model`**](#model-component): The data model that stores the data.
+* [**`Command`**](#command-component): Represents a command that the user can execute.
+* [**`Storage`**](#storage-component): Reads data from, and writes data.
```
i like the links provided here

NGXZS commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
+![Sequence Diagram](images\ArchitectureSequanceDiagram.png)
+
+### UI Component
+The UI component is responsible for parsing user input into commands that can be executed by the logic component.
+There is a `analyzeInput` method in the `Parser` class that interprets the user input and classifies 
+it into a `CommandType` enum.<br>
+The `splitInput` method is used to split the user input into an array of strings, according to the command type.
+It returns an array containing any arguments that are needed to execute the command.
```
Nit: would the explanation be more readable if we follow the format of Logic Component where each method/class has its respective explanation in its respective paragraph?

NGXZS commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
+and pass the user input to corresponding classes for analysis and execution.
+
+### Command Component
+The command component consists of two different command interfaces: `MainCommand` and `OrderCommand`. The `MainCommand`
+interface is for the various command classes that are used in the `MainLogic`, while the `OrderCommand` interface is for
+command classes used in `OrderLogic`.  
+
+A `Command` object will be created in either the `MainLogic` or `OrderLogic`
+based on what command the user has inputted(e.g., `MainHelpCommand` object is created when the user inputs the `help`
+command). The `execute()` method of the `Command` object is then called to execute the command, which may require
+certain arguments based on the type of command.
```
See nit above (similar to UI component).
Alternatively, a diagram may be useful to make it more readable/ consistent

NGXZS commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
+
+## Implementation
+
+### `MainLogic`
```
I like the consistent format in explaining the individual methods of MainLogic :)

NGXZS commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,137 @@
+and `<quantity_of_item>` is an integer of the amount of that item to be added.
+
+**Delete**  
+In `OrderLogic`, items from the current order can be removed via the
```
just a nit: to make it consistent, we may want to standardize using "Inside" or "in"

NGXZS reviewed (commented):
a few nits, but overall, clear and readable DG :))

adamzzq committed:
Fix the bug of view menu item name overflow

adamzzq committed:
Merge pull request #128 from adamzzq/branch-bugfix-menu-display

Fix the bug of view menu item name overflow

webtjs committed:
OrderHelpCommand.java: Fix incorrect command description in help message

webtjs committed:
Merge pull request #129 from webtjs/branch-bugfix-incorrect-order-help-description

OrderHelpCommand.java: Fix incorrect command description in help message

adamzzq committed:
Change menu item ID to numbers in natural order

adamzzq committed:
Fix MenuDeleteCommand's JavaDoc style

adamzzq committed:
Merge pull request #130 from adamzzq/branch-simplify-menu-item-id

Change menu item ID to numbers in natural order

adamzzq committed:
Change large values of menu items prices to scientific notation

Xb990219 committed:
Change menu/order exit from 'bye' to 'cancel', update UG and DG

Xb990219 committed:
Update complete message

Xb990219 committed:
Fix checkstyle

Xb990219 committed:
Merge pull request #131 from Xb990219/master

Update DG,UG, fix some minor bugs

webtjs committed:
Add more descriptive messages for invalid commands

webtjs committed:
Add messages for invalid commands in the different interfaces

webtjs committed:
Merge pull request #132 from webtjs/branch-bugfix-unclear-error-messages

Add more descriptive messages for invalid commands

webtjs committed:
Add indication of which interface user is currently in

webtjs committed:
Merge pull request #133 from webtjs/branch-bugfix-current-interface-unclear

Add indication of which interface user is currently in

Zhengwinter committed:
Fix documentation in User Guide to clarify organization structure

Zhengwinter committed:
Merge pull request #134 from Zhengwinter/master

Fix documentation in User Guide to clarify organization structure

webtjs committed:
Add invalid command messages for default case in switch case

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-bugfix-menu-display

webtjs committed:
Merge pull request #135 from webtjs/branch-bugfix-unclear-error-messages

Add invalid command messages for default case in switch case

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-bugfix-menu-display

adamzzq committed:
Make menu class SLAP more

adamzzq committed:
Merge pull request #136 from adamzzq/branch-bugfix-menu-display

Branch bugfix menu display

webtjs committed:
Add feature to revert changes to menu after using "cancel" command

webtjs committed:
Fix MainCommandTest.java JUnit test and MainLogic.java checkstyle

webtjs committed:
Merge pull request #137 from webtjs/branch-bugfix-revert-changes-with-cancel

Add feature to revert changes to menu after using "cancel" command

adamzzq committed:
Fix decimal point format bug

adamzzq committed:
Merge pull request #138 from adamzzq/branch-fix-decimal-point-format

Fix decimal point format bug

webtjs committed:
Fix bug when quantity of item added to order is too large

adamzzq committed:
Merge pull request #139 from webtjs/branch-bugfix-order-add-quantity-crash

Fix bug when quantity of item added to order is too large

adamzzq committed:
Show receipt instead of single items in the view order command

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-fix-view-receipt

adamzzq committed:
Merge pull request #140 from adamzzq/branch-fix-view-receipt

Branch fix view receipt

Zhengwinter committed:
Implement restaurant name, location, order taker information in order

Zhengwinter committed:
Merge branch 'master' into master

Zhengwinter committed:
Enclose order type input in try/catch block and improve robustness of input

Zhengwinter committed:
Resolve minor merge conflicts

Zhengwinter committed:
Resolve minor bug in test case

Zhengwinter committed:
Merge pull request #141 from Zhengwinter/master

Implement restaurant name, location, order taker information in order

webtjs committed:
Handle invalid inputs for restaurant details and order type

adamzzq committed:
Change receipt format

adamzzq committed:
Change receipt format

adamzzq committed:
Merge remote-tracking branch 'origin/branch-print-receipt' into branch-print-receipt

# Conflicts:
#	src/main/java/model/Order.java

adamzzq committed:
Fix checkstyle issue

adamzzq committed:
Update receipt format
* Add discount and subtotal in order summary
* Refactor code to reduce method complexity

adamzzq committed:
Add JavaDoc to new methods in Order.java

adamzzq committed:
Update JavaDoc in Order.java

adamzzq committed:
Update JavaDoc in Order.java

adamzzq committed:
Improve code quality

adamzzq committed:
Add range checking for discount

adamzzq committed:
Add range checking for discount

Zhengwinter committed:
Add Zhengwinter's PPP

Zhengwinter committed:
Merge pull request #145 from Zhengwinter/master

Add Zhengwinter's PPP

adamzzq committed:
Update line alignment according to review

adamzzq committed:
Fix charge alignment bug

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp

adamzzq committed:
Merge pull request #144 from adamzzq/branch-print-receipt

Change receipt format

webtjs committed:
Merge pull request #143 from webtjs/branch-HandleInvalidDetails

Handle invalid inputs for restaurant details and order type

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-stats

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp

webtjs committed:
Add storage feature for restaurant details, menus, and orders

webtjs committed:
Merge branch 'master' into branch-Storage

# Conflicts:
#	src/main/java/model/Order.java

webtjs committed:
MainLogic.java: Fix checkstyle error

webtjs committed:
Merge pull request #146 from webtjs/branch-Storage

Add storage feature for restaurant details, menus, and orders

webtjs committed:
Add webtjs PPP

webtjs committed:
Merge pull request #148 from webtjs/branch-webtjs-PPP

Add webtjs PPP

webtjs committed:
Storage.java: Handle NoSuchElement exception when reading save file

webtjs committed:
Merge pull request #149 from webtjs/branch-Storage

Storage.java: Handle NoSuchElement exception when reading save file

webtjs committed:
MainLogic.java: Refactor code

webtjs committed:
Restaurant.java: Update authorship

webtjs committed:
Merge pull request #150 from webtjs/master

MainLogic.java: Refactor code

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp

adamzzq committed:
Start PPP draft

adamzzq committed:
Merge pull request #151 from adamzzq/branch-ppp

Zheqi's ppp

Xb990219 committed:
Add PPP

Xb990219 committed:
Merge pull request #152 from Xb990219/master

Add PPP

Xb990219 committed:
update PPP

Xb990219 committed:
Merge pull request #153 from Xb990219/master

update PPP

webtjs committed:
Add feature to view and edit restaurant info

webtjs committed:
Merge pull request #154 from webtjs/branch-MainRestaurantInfo

Add feature to view and edit restaurant info

Zhengwinter committed:
Added documentation for various commands

Zhengwinter committed:
Merge pull request #155 from Zhengwinter/master

Added documentation for various commands

Zhengwinter committed:
Refactor code to remove unused token in initializeSystem in  MainLogic

Zhengwinter committed:
Merge pull request #156 from Zhengwinter/master

Refactor code to remove unused token in initializeSystem in  MainLogic

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-stats

Zhengwinter committed:
Add test case for menu commands and updated Zhengwinter's PPP

Zhengwinter committed:
Comment out test that failed as it is being worked on

Zhengwinter committed:
Remove menu command wildcard import

Zhengwinter committed:
Remove unused import statements in menuTest

Zhengwinter committed:
Merge pull request #160 from Zhengwinter/master

Add test case for menu commands and updated Zhengwinter's PPP

webtjs committed:
Fix DG bugs based on review by TA

webtjs committed:
Update images for edited diagrams

webtjs committed:
Abstract case if menu is not found into MainEditMenuCommand.java

adamzzq committed:
Improve menu display
* Item IDs are sorted in ascending order after each deletion
* Add a space before the price numbers

webtjs committed:
MainCommandTest.java: Update JUnit test for MainEditMenuCommand

webtjs committed:
Merge pull request #162 from webtjs/branch-MainCreateEditMenuCommand

Abstract case if menu is not found into MainEditMenuCommand.java

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-fix-menuItem-id

adamzzq committed:
Change menu related test cases according to new format

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-stats

adamzzq committed:
Merge pull request #163 from adamzzq/branch-fix-menuItem-id

Improve menu display

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-stats

Xb990219 committed:
Updatate MenuLogic in DG

Xb990219 committed:
Add documentation for all methods used

Xb990219 committed:
resolve conflict

webtjs committed:
Update model component class diagram based on PR review

Xb990219 committed:
resolve conflict-2

Xb990219 committed:
update DG with product scope

Xb990219 committed:
Merge pull request #164 from Xb990219/master

Updatate DG

webtjs committed:
Storage.java: Add documentation

webtjs committed:
Merge pull request #165 from webtjs/branch-Storage

Storage.java: Add documentation

webtjs committed:
Merge branch 'master' into branch-DG

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-stats

adamzzq committed:
Add OrderStatistics class to track restaurant performance

adamzzq committed:
Merge pull request #166 from adamzzq/branch-stats

Branch stats

adamzzq committed:
Add OrderStatistics related classes and variables
* StatsLogic: separate interface from main to handle restaurant performance related commands
* StatsCommand: implementations of view all kinds of performance data
* Add new CommandTypes to facilitate StatsLogic
* Refactor stats-related classes into packages

adamzzq committed:
Update UG for the newly added OrderStatistics feature

webtjs committed:
Update DG for v2.1

Add sequence diagram for OrderLogic, add additional manual testing in DG

adamzzq committed:
Merge pull request #168 from adamzzq/branch-StatsLogic

Add OrderStatistics related classes and variables

Zhengwinter committed:
Update community interaction in PPP

Zhengwinter committed:
Merge branch 'master' of github.com:AY2324S2-CS2113-F14-2/tp

webtjs committed:
Merge pull request #161 from webtjs/branch-DG

Fix DG bugs based on review by TA

Zhengwinter committed:
Add FAQ for net revenue vs profit

Zhengwinter committed:
Merge branch 'master' of github.com:AY2324S2-CS2113-F14-2/tp

webtjs committed:
Update UserGuide with new features (save data, restaurant info)

webtjs committed:
Merge pull request #170 from webtjs/branch-UG

Update UserGuide with new features (save data, restaurant info)

webtjs committed:
Merge branch 'master' into branch-webtjs-PPP

adamzzq committed:
Add constraints and error messages of user input values
* In menu interface: item price must be a number in (0, 10000); item name cannot be empty string
* Make it case-insensitive in checking items with the same name (e.g. ChIcKeN == cHICken )
* In each order, quantity for an added item must be an integer in (0, 10000) and total count capped at 9999

adamzzq committed:
Merge pull request #171 from adamzzq/branch-quantity-limit

Add constraints and error messages to user input values

adamzzq committed:
Add discount to menu's complete command

adamzzq committed:
Merge pull request #172 from adamzzq/branch-quantity-limit

Add discount to menu's complete command

Zhengwinter committed:
Merge branch 'master' of github.com:AY2324S2-CS2113-F14-2/tp

Zhengwinter committed:
Merge pull request #173 from Zhengwinter/master

Edit user guide FAQ

Xb990219 committed:
update UG and DG

webtjs committed:
Update PPP

Xb990219 committed:
Merge pull request #174 from Xb990219/master

update UG and DG

Zhengwinter committed:
Update UG and rectify indexing issue

Zhengwinter committed:
Resolve merge conflicts

Zhengwinter committed:
Merge pull request #175 from Zhengwinter/master

Update UG and DG

webtjs committed:
Storage.java: Fix bug relating to save file

Fix issue where users are able to enter negative values for price and
item ID in orders.txt and menus.txt

adamzzq committed:
Change the display layout for view item command in the order interface

webtjs committed:
Merge pull request #176 from webtjs/master

Storage.java: Fix bug relating to save file

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp

adamzzq committed:
Merge pull request #177 from adamzzq/branch-view-items

Branch view items

adamzzq committed:
Add some test cases for OrderStatistics

adamzzq committed:
Fix bugs:
* BestSelling command when no orders in the orderList
* Discount total price not correctly shown in Main interface

webtjs committed:
Update example menu IDs in DeveloperGuide

webtjs committed:
Merge pull request #178 from webtjs/master

Update example menu IDs in DeveloperGuide

adamzzq committed:
Fix bugs:
* BestSelling command when no orders in the orderList
* Discount total price not correctly shown in Main interface
* long userName/restaurantName/restaurantAddress issue. User name
is capped at 50 chars and restaurantName/restaurantAddress can wrap to the next line if too long

adamzzq committed:
Merge pull request #179 from adamzzq/branch-fix-receipt-bugs

Branch fix receipt bugs

webtjs committed:
MainCommandTest.java: Add more JUnit tests

adamzzq committed:
Fix bug: View menu command should show "menu <menu_id> with XX items"

webtjs committed:
Merge pull request #180 from webtjs/master

MainCommandTest.java: Add more JUnit tests

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-fix-view-menu-display

adamzzq committed:
Merge pull request #181 from adamzzq/branch-fix-view-menu-display

Fix bug: View menu command should show "menu <menu_id> with XX items"

Xb990219 committed:
fix images problem in DG

Xb990219 committed:
fix images problem in DG

Xb990219 committed:
Merge pull request #182 from Xb990219/master

Fix image problem in DG

Xb990219 committed:
finalised UG

Xb990219 committed:
Merge pull request #183 from Xb990219/master

finalised UG

Zhengwinter committed:
Update order command test cases

Zhengwinter committed:
Add test cases for all order commands

Zhengwinter committed:
Fix format issue in test case

Zhengwinter committed:
Fix minor grammar issue

Zhengwinter committed:
Merge pull request #184 from Zhengwinter/master

Add test cases for all order commands

webtjs committed:
Add Storage component in DeveloperGuide

webtjs committed:
Update Storage component in DG

webtjs committed:
Merge pull request #185 from webtjs/master

Add Storage component in DeveloperGuide

adamzzq committed:
Add StatsLogic in DG

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-statsLogic-DG

# Conflicts:
#	src/main/java/command/order/OrderDeleteCommand.java

adamzzq committed:
Update README.md to add PPP link

adamzzq committed:
Restore OrderDeleteCommand.java edit by removing a space

adamzzq committed:
Merge pull request #186 from adamzzq/branch-statsLogic-DG

Branch stats logic dg

Zhengwinter committed:
Update Zhengwinter's PPP

Zhengwinter committed:
Merge pull request #187 from Zhengwinter/master

Update Zhengwinter's PPP

Xb990219 committed:
Finalised DG

Xb990219 committed:
Merge pull request #188 from Xb990219/master

Finalised DG

webtjs committed:
Update Storage related components

Storage now loads order and menu data regardless of whether a
restaurant save data exists or not

webtjs committed:
Merge branch 'master' into branch-RefactorFor2.1

# Conflicts:
#	src/main/java/logic/MainLogic.java

webtjs committed:
Merge pull request #189 from webtjs/branch-RefactorFor2.1

Branch refactor for v2.1

webtjs committed:
Decrease image size in DG

webtjs committed:
Merge pull request #190 from webtjs/master

Decrease image size in DG

webtjs committed:
Add pagebreak in DG

webtjs committed:
Merge pull request #191 from webtjs/master

Add pagebreak in DG

webtjs committed:
Change pagebreak in DG

webtjs committed:
Update grammatical error in UG

webtjs committed:
Merge pull request #192 from webtjs/master

Change pagebreak in DG

webtjs committed:
Add additional pagebreak in DG

webtjs committed:
Merge pull request #193 from webtjs/master

Add additional pagebreak in DG

webtjs committed:
Update DG

webtjs committed:
Change image source file in DG

webtjs committed:
Merge pull request #194 from webtjs/master

Update DG

webtjs committed:
Fix formatting issue in DG

webtjs committed:
Merge pull request #195 from webtjs/master

Fix formatting issue in DG

webtjs committed:
Merge branch 'master' into branch-webtjs-PPP

adamzzq committed:
Update Zheqi's PPP

adamzzq committed:
Merge pull request #196 from adamzzq/branch-ppp

Update Zheqi's PPP

adamzzq committed:
Add new User Stories in DG, and correct some grammatical errors

adamzzq committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F14-2/tp into branch-more-user-stories

adamzzq committed:
Merge pull request #197 from adamzzq/branch-more-user-stories

Branch more user stories

adamzzq committed:
Update Zheqi's PPP

adamzzq committed:
Update Zheqi's PPP

adamzzq committed:
Merge pull request #198 from adamzzq/branch-ppp

Update Zheqi's PPP

webtjs committed:
Update webtjs PPP

webtjs committed:
Merge branch 'master' into branch-webtjs-PPP

webtjs committed:
Update LoC in webtjs PPP

Zhengwinter committed:
Edit UG and fix grammatical errors and formatting issues

Zhengwinter committed:
Merge pull request #200 from Zhengwinter/master

Edit UG and fix grammatical errors and formatting issues

webtjs committed:
Merge pull request #199 from webtjs/branch-webtjs-PPP

Update webtjs PPP

Zhengwinter committed:
Comment out loggers and update DG

webtjs committed:
Update link in webtjs PPP

Zhengwinter committed:
Remove unused imports

webtjs committed:
Merge pull request #202 from webtjs/branch-webtjs-PPP

Update link in webtjs PPP

Zhengwinter committed:
Merge pull request #201 from Zhengwinter/master

Comment out loggers and update DG

Zhengwinter committed:
Update Wentao and Xiao Bo PPP

Zhengwinter committed:
Merge pull request #203 from Zhengwinter/master

Update Wentao and Xiao Bo PPP

adamzzq committed:
Update Zheqi's PPP

adamzzq committed:
Merge pull request #204 from adamzzq/branch-ppp

Update Zheqi's PPP

Zhengwinter committed:
Minor update to PPP

Zhengwinter committed:
Merge pull request #205 from Zhengwinter/master

Minor update to PPP
